src/services/offer.js

```diff
@@ -64,6 +64,12 @@ class OfferService {
         ).map((o) => o.withCompany()));
 
     }
+
+    async getOfferById(id) {
+        const offer = await Offer.findById(id);
+        return offer;
+    }
```

Comment:
This can be simplified since you can return the find promise directly:

```js
getOfferById(id) {return Offer.findById(id)}
```

This works since `Offer.findById` returns a promise that resolves to the found value. You don't need the `async/await` in this case since you are not waiting for the resolving in this function, but actually in the function that calls `OfferService.getOfferById`, so you are just creating this intermediary function. LifeProTip: Never wait for a promise when you don't need to read its value.